### PR TITLE
Fix function parameters parsing without implementations

### PR DESCRIPTION
**Pull request by NeilKleistGao**

### CODE REVIEW

**Comment by NeilKleistGao** [comment]
Also, fix `=>` right-association

**Comment by NeilKleistGao** [comment]
So far so good, but I'm not very sure about what to do next step. Any ideas please? :open_mouth:

**Comment by LPTK** [comment]
Let's merge it ASAP. No need to let it linger any longer. Request a review and mark it as ready when you are.

File: shared/src/main/scala/mlscript/helpers.scala [hunk 1]
```diff
@@ -422,8 +422,10 @@ trait TermImpl extends StatementImpl { self: Term =>
       case p: TypeName => AppliedType(p, rhs.toType_! :: Nil)
       case _ => throw new NotAType(this)
     }
+    case Tup(fields) => Tuple(fields.map(fld => (fld._1, fld._2 match {
+      case Fld(_, _, v) => Field(N, v.toType_!)
```
Comment (LPTK) [inline shared/src/main/scala/mlscript/helpers.scala]: If the field is mutable we should make it like this:
```suggestion
      case Fld(m, s, v) => val ty = v.toType_!; Field(Option.when(m)(ty), ty)
```

**Review by LPTK — APPROVED** [review]
Great, thanks for the quick fixes!

**Comment by LPTK** [comment]
After addressing my comment, you can go ahead and use the "squash and merge". Don't forget to clean up the commit message (remove the useless lines).

**Comment by NeilKleistGao** [comment]
> After addressing my comment, you can go ahead and use the "squash and merge". Don't forget to clean up the commit message (remove the useless lines).

But I got no write access because I'm not in the organization yet. 🤣

**Pull request merged**